hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -210,15 +213,41 @@ private HoodieLogBlock readBlock() throws IOException {
   private HoodieLogBlock createCorruptBlock() throws IOException {
     log.info("Log " + logFile + " has a corrupted block at " + inputStream.getPos());
     long currentPos = inputStream.getPos();
-    long nextBlockOffset = scanForNextAvailableBlockOffset();
+    byte[] corruptedBytes;
+    int corruptedBlockSize = 0;
+    boolean done = true;
+    // read upto logblocksize to find next magic header
+    do {
+      corruptedBytes = new byte[logBlockSize];
+      int numberOfBytesRead;
+      try {
+        inputStream.readFully(corruptedBytes, 0, logBlockSize);
+        numberOfBytesRead = logBlockSize;
+      } catch (EOFException e) {
+        // in case unable to read logBlockSize worth of bytes
+        inputStream.seek(currentPos);
+        numberOfBytesRead = inputStream.available();
```

Comment:
why not use `available()` to estimate actual bytes to begin with, instead of course correcting in the catch block?

Reply:
The reason to do this was because of the contract of available(..) mentioned here https://docs.oracle.com/javase/7/docs/api/java/io/FilterInputStream.html#available() and how we want to read minimum of default size. But I changed this using the BufferedInputStream so probably no need to discuss further..